br/pkg/lightning/restore/restore.go

```diff
@@ -658,6 +664,24 @@ loop:
 				err = sqlWithRetry.Exec(worker.ctx, "run create schema job", stmt)
 				task.End(zap.ErrorLevel, err)
 				if err != nil {
+					// try to imitate IF NOT EXISTS behavior for parsing errors
+					var (
+						exists = false
+						err2   error
+					)
+
+					switch job.stmtType {
+					case schemaCreateDatabase:
+						exists, err2 = common.SchemaExists(worker.ctx, session, job.dbName)
+					case schemaCreateTable:
+						exists, err2 = common.TableExists(worker.ctx, session, job.dbName, job.tblName)
+					}
+					if err2 != nil {
+						task.Error("failed to check table exists", zap.Error(err2))
+					}
+					if exists {
```

Comment:
Table existence has been checked in `makeJobs`. So I don't think there should be this error here.

Reply:
https://github.com/pingcap/tidb/pull/38327/commits/a6fafec703d4345a5e4ec63b5c3275cfc5242fe3

only check the error in "database" branch now

Reply:
In line 665, the log task is end with an error. Since we consider ignoring the already exists error, I think we can let it end with `nil`.

Reply:
[2cd6974](https://github.com/pingcap/tidb/pull/38327/commits/2cd697406ef54f6eef3f96a110bf1fe570fb5ed0)